src/nnet3/nnet-diagnostics.h

```diff
@@ -97,6 +97,23 @@ class NnetComputeProb {
   // or NULL if there is no such info.
   const SimpleObjectiveInfo *GetObjective(const std::string &output_name) const;
 
+  // return string of info which summarized the objectives for different outputs.
+  std::string GetAllObjectiveInfo(double* tot_weight) const {
+    std::ostringstream ostr;
+    std::string obj_info_str = ostr.str();
+    unordered_map<std::string, SimpleObjectiveInfo,
+      StringHasher>::const_iterator objf_it = objf_info_.begin(),
+      objf_end = objf_info_.end();
+    for(; objf_it != objf_end; ++objf_it) {
+      double objf_per_frame = objf_it->second.tot_objective / objf_it->second.tot_weight;
+      ostr << " Objf per frame for '" << objf_it->first
```

Comment:
it is used in nnet3-show-progress. I think the message is consistent. What is the problem?